session_rpcserver.go

```diff
@@ -88,12 +95,49 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
 	}
 	if sess.Expiry.Before(time.Now()) {
 		log.Debugf("Not resuming session %x with expiry %s",
-			sess.LocalPublicKey.SerializeCompressed(), sess.Expiry)
+			localPubKey, sess.Expiry)
+
+		if err := s.db.RevokeSession(pubKey); err != nil {
+			return fmt.Errorf("error revoking session: %v", err)
+		}
+
 		return nil
 	}
 
 	authData := []byte("Authorization: Basic " + s.basicAuth)
-	return s.sessionServer.StartSession(sess, authData)
+	if err = s.sessionServer.StartSession(sess, authData); err != nil {
+		return err
+	}
+
+	go func() {
+		sessionClosedSub := s.sessionServer.SubscribeToSessionClosed(
+			pubKey,
+		)
+		go func() {
```

Comment:
Do we really need a second goroutine here? Can't we just continue in the same one?

Reply:
oops. mistake 🙈